### ISSUE TITLE
table of contents for note

### ISSUE THREAD

**Issue by trialley** [zadam/trilium#2603]
### Describe feature

when will trilium support catalogue for notes?

**Comment by FliegendeWurst** [comment]
What kind of catalogue?

**Comment by trialley** [comment]
> What kind of catalogue?

like this:
![image](https://user-images.githubusercontent.com/13725220/153343565-415fedf2-8e47-4ea7-9d10-1dc510c069cb.png)

**Comment by vulbox** [comment]
Trilium has a diary feature, but also wants to have a side column for long articles.
Thank you for your great efforts in developing this software.
Like this 
![图片](https://user-images.githubusercontent.com/69760521/154209806-a759c184-4e0d-4dfd-9c34-ebabdf2a4a83.png)

**Comment by trialley** [comment]
Hi @zadam, like @vulbox said, we need a catalogue for long articles.

I have tried to change the source code for adding catalogue, but it's a little difficult for me.

![image](https://user-images.githubusercontent.com/13725220/154217126-60471653-404a-4eda-bed4-e3b85ad129f7.png)

**Comment by zadam** [comment]
Oh, so you mean Table of Contents. Closing in favor of https://github.com/zadam/trilium/issues/533

**Comment by trialley** [comment]
It's content, not catalogue, my bad.

It seems that this feature has been developed for 3-4 years in [ckeditor5](https://github.com/ckeditor/ckeditor5). Let me see if I can push it forward.

Thanks!

**Comment by trialley** [comment]
Hi @zadam , [ckeditor5](https://github.com/ckeditor/ckeditor5) already support [minimap](https://ckeditor.com/docs/ckeditor5/latest/api/minimap.html).
Will trilium support this feature?

**Comment by zadam** [comment]
Minimap is a different feature from ToC so it would be better to discuss it separately.

[Demo](https://ckeditor.com/docs/ckeditor5/latest/features/minimap.html#demo) doesn't work, it also says "This is a feature preview, and as such it is not recommended for production use."). I'm also not sure if this would be a fit. So no plans ATM.

**Comment by vulbox** [comment]
> Hi @zadam , [ckeditor5](https://github.com/ckeditor/ckeditor5) already support [minimap](https://ckeditor.com/docs/ckeditor5/latest/api/minimap.html). Will trilium support this feature?

Do you have any new progress on this function? It seems that Ckeditor5 has no intention to update the TOC function. Please feel free to contact me if you need help, although my programming skills are very limited.

**Comment by zadam** [comment]
No, I don't plan to work on this on my own and would probably not accept a 3rd party solution either. I'm waiting on the upstream implementation.
